test/winter_onboarding_2021/fleet_management/handlers/cab_test.clj

```diff
@@ -1,31 +1,59 @@
 (ns winter-onboarding-2021.fleet-management.handlers.cab-test
   (:require [clojure.test :refer [deftest is testing use-fixtures]]
-            [winter-onboarding-2021.fleet-management-service.handlers.cab :as cab]
+            [clojure.string :as str]
+            [winter-onboarding-2021.fleet-management-service.handlers.cab :as handlers]
+            [winter-onboarding-2021.fleet-management-service.views.cab :as views]
+            [winter-onboarding-2021.fleet-management-service.models.cab :as models]
+            [winter-onboarding-2021.fleet-management-service.views.layout :as layout]
             [winter-onboarding-2021.fleet-management.fixtures :as fixtures]))
 
 (use-fixtures :once fixtures/config fixtures/db-connection)
 (use-fixtures :each fixtures/clear-db)
 
 (deftest add-cab
-  (testing "POST  /cabs/ endpoint with valid cab data, shoudld redirect to '/cabs/new?success=true' "
+  (testing "POST /cabs/ endpoint with valid cab data, shoudld redirect to '/cabs/<<uuid of new cab>> "
+    (let [response (handlers/create {:multipart-params
+                                     {:name "Test cab"
+                                      :licence_plate "KA20X1234"
+                                      :distance_travelled "1223"}})]
+      (is (= 302 (:status response)))
+
+      (is (= {:success true
+              :style-class "alert alert-success"
+              :message "Cab added successfully!"}
+             (:flash response)))
+
+      (is (= true (-> response
```

Comment:
What is this test doing?

Reply:
It's making sure that the `Location` header contains a string of length 36 characters after the second `/`.

So, if we have `/cabs/73f06e14-fdd9-4c0e-b88b-3ad2202eb21f` in `Location` header then it will make sure we have the UUID after the second `/` .

The whole process could be better if we use a regex, but I couldn't figure it out for this pattern.

Reply:
@shhivam what the test was trying to do was not obvious to me. I would suggest 2 things to improve readability: 
- coercing the location header to an URL, either using `as-url` or https://github.com/cemerick/url and avoid string manipulation
- There are ways to check if the id is an UUID, but that's not helpful as a test. What you want to test is that you're being redirected to the correct show page. You can get id of the created cab from the db using the license number, and compare it with the id in the redirect location.

Reply:
@govindkrjoshi I have implemented the second point of actually querying the DB to see if the cab was added or not.
Regarding the use of `as-url` or https://github.com/cemerick/url, I could not find a straightfoward way to use them without string manipulation.

For example, usiing `as-url`:

```clojure
(-> (as-url "http://localhost:3000/" (:location headers)
     :path
     (= (str "/cabs/" (:cabs/id cab-from-db))
```